### ISSUE TITLE
Should we require `this.` on calls to sibling `init()` routines?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#23031]
As Chapel is defined today, initializers can only be called from within other initializers, and tend to be thought of as taking two forms:
* `this.init(...);` -- call a sibling initializer (typically for code reuse / to avoid duplication)
* `super.init(...);` -- call my superclass's initializer

We also support calls to `init(...)` without a receiver, which defaults to `this`, like other method calls, since we are within a method:

* `init(...);` -- equivalent to `this.init(...);`

Generally speaking, stylistically, I think most of us tend to use the explicit `this.init(...);` form maybe for clarity and maybe for symmetry with other initializations, which tend to be of the form `this.x = ...;`, `this.y = ...;`, etc.

This issue asks whether we would want to stop supporting this receiver-less form, where reasons for doing so might be:
* to make the receiver super-clear / avoid any implications that we're calling a `super` initializer or some other `init` procedure or method
* to avoid the potential for ambiguity if we adopted `init;` as the replacement for `this.complete();`  (https://github.com/chapel-lang/chapel/issues/23005)

Another approach would be to continue to support both forms, have our style guide prefer the `this.init(...);` form, and have linter-like tools flag `init(...);` calls.

**Comment by lydia-duncan** [comment]
> * to avoid the potential for ambiguity if we adopted `this;` as the replacement for `this.complete();`

I think you meant:

> * to avoid the potential for ambiguity if we adopted `init;` as the replacement for `this.complete();`

**Comment by bradcray** [comment]
I can't call myself a champion of this issue, particularly now that @DanilaFe implemented `init this;` as the replacement for `this.complete();` in https://github.com/chapel-lang/chapel/pull/23283.  Specifically, I think it makes sense that method calls don't normally need a `this.` and don't think we should make an exception for `init(...);` particularly now that we've moved away from `this.complete();` alternatives that would be most likely to result in confusion.  Closing this issue as a result (but someone else can re-open if they want to champion it).
